c7n/policy.py

```diff
@@ -73,8 +75,13 @@ def __init__(self, data, options):
         # We store all the policies passed in so we can refilter later
         self._all_policies = []
         for p in self.data.get('policies', []):
-            self._all_policies.append(
-                Policy(p, options, session_factory=self.test_session_factory()))
+            if 'all' in options.regions:
+                options.regions = utils.get_available_regions(p['resource'])
```

Comment:
session creation is expensive, so i'd scratch the utility method here, this would probably be nicer as function that can take the session and policy set/collection and return back an expanded one.